pkg/helm/client.go

```diff
@@ -526,80 +536,17 @@ func (h *HelmClient) getRESTClientGetter(namespace string) genericclioptions.RES
 	return cfgFlags
 }
 
-// cleanUpGenericMap is a helper to "cleanup" generic yaml parsing where nested maps
-// are unmarshalled with type map[interface{}]interface{}
-func cleanUpGenericMap(in map[string]interface{}) map[string]interface{} {
-	result := make(map[string]interface{})
-	for k, v := range in {
-		result[fmt.Sprintf("%v", k)] = cleanUpMapValue(v)
-	}
-	return result
-}
-
-// Cleans up the value in the map, recurses in case of arrays and maps
-func cleanUpMapValue(v interface{}) interface{} {
-	// Keep null values as nil to avoid type mismatches
-	if v == nil {
-		return nil
-	}
-	switch v := v.(type) {
-	case []interface{}:
-		return cleanUpInterfaceArray(v)
-	case []map[string]interface{}:
-		return cleanUpGenericMapArray(v)
-	case []map[interface{}]interface{}:
-		return cleanUpInterfaceMapArray(v)
-	case map[string]interface{}:
-		return cleanUpGenericMap(v)
-	case map[interface{}]interface{}:
-		return cleanUpInterfaceMap(v)
-	case string:
-		return v
-	case int:
-		return v
-	case bool:
-		return v
-	case float64:
-		return v
-	default:
-		return fmt.Sprintf("%v", v)
-	}
-}
-
-// Cleans up a slice of interfaces into slice of actual values
-func cleanUpInterfaceArray(in []interface{}) []interface{} {
-	result := make([]interface{}, len(in))
-	for i, v := range in {
-		result[i] = cleanUpMapValue(v)
-	}
-	return result
-}
-
-// Cleans up a slice of map to interface into slice of actual values
-func cleanUpGenericMapArray(in []map[string]interface{}) []map[string]interface{} {
-	result := make([]map[string]interface{}, len(in))
-	for i, v := range in {
-		result[i] = cleanUpGenericMap(v)
-	}
-	return result
-}
-
-// Cleans up a slice of map to interface into slice of actual values
-func cleanUpInterfaceMapArray(in []map[interface{}]interface{}) []map[string]interface{} {
-	result := make([]map[string]interface{}, len(in))
-	for i, v := range in {
-		result[i] = cleanUpInterfaceMap(v)
+func cleanUpGenericMap(m map[string]interface{}) (map[string]interface{}, error) {
+	b, err := yaml.Marshal(m)
```

Comment:
why not k8syaml?

Reply:
i think because k8syaml cannot marshal a map[interface{}]interface{}. i will test and add a comment

Reply:
i added comments to clarify